physical/postgresql/postgresql.go

```diff
@@ -236,6 +237,56 @@ func NewPostgreSQLBackend(conf map[string]string, logger log.Logger) (physical.B
 		upsert_function: quoted_upsert_function,
 	}
 
+	// Determine if we should create tables.
+	raw_skip_create_table, ok := conf["skip_create_table"]
+	if !ok {
+		raw_skip_create_table = "false"
+	}
+	skip_create_table, err := parseutil.ParseBool(raw_skip_create_table)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse value for `skip_create_table`: %w", err)
+	}
+	if !skip_create_table {
+		txn, err := db.BeginTx(context.TODO(), &sql.TxOptions{
+			Isolation: sql.LevelRepeatableRead,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to begin transaction to attempt table creation: %w", err)
+		}
+
+		defer txn.Rollback()
+
+		createTableQuery := "CREATE TABLE IF NOT EXISTS " + quoted_table + " (" +
+			`parent_path TEXT COLLATE "C" NOT NULL,` +
+			`  path        TEXT COLLATE "C",` +
+			`  key         TEXT COLLATE "C",` +
+			`  value       BYTEA,` +
+			`  CONSTRAINT pkey PRIMARY KEY (path, key)` +
+			`);`
+		if _, err := db.Exec(createTableQuery); err != nil {
```

Comment:
The one issue with this is that, if we don't have table creation permissions, the query will fail even if the tables already exist. I think this could be a problem in some deployments, but PostgreSQL was not part of 2.0.0's GA compatibility guarantee so I think we can assume fresh deployments from here on out. They'll need to manually set `skip_create_tables=false` in their configuration.

Reply:
Is there a secure way for providing OpenBao with  PostgreSQL credentials (including for allowing table creation) without exposing those credentials in the configuration files ?
Generally spiking, this seems to me another "chicken and egg problem" that we will not solve here, but we could maybe consider in the future the possibility to have in OpenBao a kind of initial secrets (bootstrap secrets) that would be available just after the unseal but before setting up the storage (maybe provided by a KMS, or by doing a ECDH ...) .
In any case, it seems to be a non-trivial problem.

Reply:
The issue with the PostgreSQL engine is that it stores all data in a single table (`table`, usually set to `openbao_kv_data`). If we want to change that (to improve scalability), we need clients to grant table creation to OpenBao's credentials, e.g., to create separate tables for each namespace or something, that are used online and not one-off at the first startup.

The user can also always start OpenBao, have it create tables, and then restart it with more limited credentials as well.